File: triptracking-library/capacitor_plugin/android/src/main/java/com/carmd/triptracking/capacitor/triptrackingCapacitorPlugin.kt
package com.carmd.triptracking.capacitor

import com.carmd.triptracking.triptracking
import com.getcapacitor.Plugin
import com.getcapacitor.PluginCall
import com.getcapacitor.PluginMethod
import com.getcapacitor.annotation.CapacitorPlugin
import com.getcapacitor.JSObject

@CapacitorPlugin(name = "triptracking")
class triptrackingCapacitorPlugin : Plugin() {

    @PluginMethod
    fun doSomething(call: PluginCall) {
        val input = call.getString("input") ?: ""
        val result = triptracking().doSomething(input)
        val ret = JSObject()
        ret.put("value", result)
        call.resolve(ret)
    }
}

File: triptracking-library/capacitor_plugin/ios/Plugin/triptrackingCapacitorPlugin.swift
import Capacitor
import triptracking

@objc(triptrackingCapacitorPlugin)
public class triptrackingCapacitorPlugin: CAPPlugin {

    @objc func doSomething(_ call: CAPPluginCall) {
        let input = call.getString("input") ?? ""
        let result = triptracking().doSomething(input: input)
        call.resolve(["value": result])
    }
}
